src/Storefront/Resources/app/storefront/test/plugin-system/plugin.manager.test.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * @jest-environment jsdom
+ */
+import PluginManager from 'src/plugin-system/plugin.manager';
+import Plugin from 'src/plugin-system/plugin.class';
+import Iterator from "../../src/helper/iterator.helper";
+
+class FooPlugin extends Plugin {
+    init() {}
+}
+
+beforeEach(() => {
+    document.body.innerHTML = '<div data-plugin="true" class="test-class"></div><div id="test-id"></div>';
+});
+
+describe('Plugin manager', () => {
```

Comment:
I counted 5 return statements in `_queryElements` but only three tests. That means you could not have all cases covered

Reply:
I considered the first one an early break for the last return :-)
But you're right, this could be seen as you did it.
So, I added a clear additional test-case `should initialize plugin with mixed selector (class and data-attribute)`